src/book/mod.rs

```diff
@@ -133,69 +137,44 @@ impl MDBook {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.get_destination());
-                fs::create_dir_all(self.get_destination())?;
-            }
-
-
-            if !self.config.get_source().exists() {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.config.get_source());
-                fs::create_dir_all(self.config.get_source())?;
+            let root = self.config.get_root();
+            let dest = self.get_destination();
+            let src = self.config.get_source();
+
+            let necessary_folders = &[root, dest, src];
+            
+            for folder in necessary_folders {
+                if !folder.exists() {
+                    fs::create_dir_all(folder)?;
+                    debug!("{} created", folder.display());
+                }
             }
 
-            let summary = self.config.get_source().join("SUMMARY.md");
+            let summary = src.join("SUMMARY.md");
 
             if !summary.exists() {
+                debug!("[*]: Creating SUMMARY.md");
 
-                // Summary does not exist, create it
-                debug!("[*]: {:?} does not exist, trying to create SUMMARY.md", &summary);
                 let mut f = File::create(&summary)?;
 
-                debug!("[*]: Writing to SUMMARY.md");
-
                 writeln!(f, "# Summary")?;
-                writeln!(f, "")?;
+                writeln!(f)?;
                 writeln!(f, "- [Chapter 1](./chapter_1.md)")?;
             }
-        }
 
-        // parse SUMMARY.md, and create the missing item related file
-        self.parse_summary()?;
+            let ch_1 = src.join("chapter_1.md");
+            if !ch_1.exists() {
+                debug!("[*] Creating {}", ch_1.display());
 
-        debug!("[*]: constructing paths for missing files");
-        for item in self.iter() {
-            debug!("[*]: item: {:?}", item);
-            let ch = match *item {
-                BookItem::Spacer => continue,
-                BookItem::Chapter(_, ref ch) |
-                BookItem::Affix(ref ch) => ch,
-            };
-            if !ch.path.as_os_str().is_empty() {
-                let path = self.config.get_source().join(&ch.path);
-
-                if !path.exists() {
-                    if !self.create_missing {
-                        return Err(format!("'{}' referenced from SUMMARY.md does not exist.", path.to_string_lossy())
-                                       .into());
-                    }
-                    debug!("[*]: {:?} does not exist, trying to create file", path);
-                    ::std::fs::create_dir_all(path.parent().unwrap())?;
-                    let mut f = File::create(path)?;
-
-                    // debug!("[*]: Writing to {:?}", path);
-                    writeln!(f, "# {}", ch.name)?;
-                }
+                let mut f = File::create(&ch_1)?;
+                writeln!(f, "# Chapter 1")?;
```

Comment:
`init` is more than a little bloated. I'm not sure if we should not refactor some of its code into separate methods ie `create_missing_*`

Reply:
Good idea, the `init` method looks a lot cleaner now.